plugins/http_plugin/http_plugin.cpp

```diff
@@ -271,6 +272,13 @@ namespace eosio {
             boost::filesystem::path sock_path = options.at("unix-socket-path").as<string>();
             if (sock_path.is_relative())
                sock_path = app().data_dir() / sock_path;
+            // The maximum length of the socket path is defined by sockaddr_un::sun_path. On Linux,
+            // according to unix(7), it is 108 bytes. On FreeBSD, according to unix(4), it is 104 bytes.
+            const auto unix_socket_limit = sizeof(sockaddr_un::sun_path);
+            if (sock_path.string().size() >= unix_socket_limit) {
+               // try a relative path which may be shorter.
+               sock_path = boost::filesystem::relative(sock_path);
```

Comment:
Probably just wait until your std::filesystem is merged before doing this, such that you don't need to change boost::filesystem to std::filesystem one more time.

You should still check if the relative path is longer than  unix_socket_limit, and log you are trying a relative path.

Reply:
If the relative is longer than the limit, we would get a system exception with a 'path is too long` description already. Extra checking doesn't do anything better.